### PR TITLE
fix(route): taptap, topic failed to respond

### PR DESCRIPTION
**Pull request by ETiV**
## 该 PR 相关 Issue / Involved issue

No issues yet, but URL: `https://rsshub.app/taptap/topic/2301` will be failed.

## 完整路由地址 / Example for the proposed route(s)

```routes
/taptap/topic/2301
/taptap/topic/2301/feed
/taptap/topic/2301/feed/created
/taptap/topic/2301/feed/created/zh_CN
/taptap/topic/2301/feed/created/en_US
/taptap/topic/2301/feed/commented
/taptap/topic/2301/feed/commented/zh_CN
/taptap/topic/2301/feed/commented/en_US
/taptap/topic/2301/elite
/taptap/topic/2301/elite/created
/taptap/topic/2301/elite/created/zh_CN
/taptap/topic/2301/elite/created/en_US
/taptap/topic/2301/elite/commented
/taptap/topic/2301/elite/commented/zh_CN
/taptap/topic/2301/elite/commented/en_US
/taptap/topic/2301/official
/taptap/topic/2301/official/created
/taptap/topic/2301/official/created/zh_CN
/taptap/topic/2301/official/created/en_US
/taptap/topic/2301/official/commented
/taptap/topic/2301/official/commented/zh_CN
/taptap/topic/2301/official/commented/en_US
/taptap/topic/2301/video
/taptap/topic/2301/video/created
/taptap/topic/2301/video/created/zh_CN
/taptap/topic/2301/video/created/en_US
/taptap/topic/2301/video/commented
/taptap/topic/2301/video/commented/zh_CN
/taptap/topic/2301/video/commented/en_US
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [ ] 新的路由 New Route
  - [ ] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

**Comment by ETiV** [comment]
（吐个槽）

[这里](https://github.com/DIYgod/RSSHub/blob/master/.github/PULL_REQUEST_TEMPLATE.md?plain=1#L17) `route` 少了个 s。

而且它在注释内部，Create PR 的编辑器没有语法着色，根本看不出来 🤣

最好让它变成 `routes（这里是注释。请在下方注释外部添加路由）`

image

File: lib/v2/taptap/topic.js [hunk 1]
```diff
@@ -33,22 +33,29 @@ module.exports = async (ctx) => {
     const topics_list_response = await got(url);
     const topics_list = topics_list_response.data;
 
-    const app_img = topics_list.data.list[0].moment.app.icon.url;
-    const app_name = topics_list.data.list[0].moment.app.title;
+    let app_img = null;
+    let app_name = null;
 
     const out = topics_list.data.list.map(async (list) => {
+        if (app_img === null && app_name === null) {
```
Comment (TonyRL) [inline lib/v2/taptap/topic.js:40]: This condition is always true. You can skip the outer `if`

File: lib/v2/taptap/utils.js [hunk 2]
```diff
@@ -35,6 +35,7 @@ const textPost = async (appId, topicId, lang = 'zh_CN') => {
 const videoPost = (video) =>
     art(path.join(__dirname, 'templates/videoPost.art'), {
         intro: video?.intro?.text,
+        previewUrl: video?.video_resource.preview_animation.url,
         videoUrl: video?.url,
         posterUrl: video?.raw_cover.url,
```
Comment (TonyRL) [inline lib/v2/taptap/utils.js:40]:
```suggestion
        previewUrl: video?.video_resource.preview_animation.original_url,
```

- `original_url` has a higher resolution
- remove unused property

File: lib/v2/taptap/topic.js (hunk 1 above)
Comment (ETiV) [reply lib/v2/taptap/topic.js:40]: >This condition is always true.

No.

I've changed app_img/app_name from a constant into a variable, and it's outside the `.map(...)` loop.

once those two variables has been set properly, they shouldn't be set again. so there is an `if` here.

**Pull request merged**